.github/workflows/production-build.yaml

```diff
@@ -29,7 +42,14 @@ jobs:
       - name: Build and push images
         working-directory: ./docker
         run: |
-          VERSION=${{ github.event.release.tag_name }} docker-compose -f docker-compose.build.yaml build --no-cache ${{ matrix.service_name }}
-          docker tag unstract/${{ matrix.service_name }}:${{ github.event.release.tag_name }} unstract/${{ matrix.service_name }}:latest
-          docker push unstract/${{ matrix.service_name }}:${{ github.event.release.tag_name }}
-          docker push unstract/${{ matrix.service_name }}:latest
+            if [ "${{ github.event_name }}" = "release" ]; then
+              echo "This workflow was triggered by a release event."
+              VERSION=${{ github.event.release.tag_name }} docker-compose -f docker-compose.build.yaml build --no-cache ${{ matrix.service_name }}
+              docker tag unstract/${{ matrix.service_name }}:${{ github.event.release.tag_name }} unstract/${{ matrix.service_name }}:latest
+              docker push unstract/${{ matrix.service_name }}:${{ github.event.release.tag_name }}
+              docker push unstract/${{ matrix.service_name }}:latest
+            else
+              echo "This workflow was triggered manually."
+              VERSION=${{ github.event.inputs.tag }} docker-compose -f docker-compose.build.yaml build --no-cache ${{ matrix.service_name }}
+               docker push unstract/${{ matrix.service_name }}:${{ github.event.inputs.tag }}
+            fi
```

Comment:
@johnyrahul why are we keeping 2 separate logic for building the docker? We should only have single logic. Can you refactor this. 

Keep the original logic as such.

Instead of directly using `${{ github.event.release.tag_name }}` or `${{ github.event.inputs.tag }}` try to assign this to a variable and then use it.

Reply:
@ritwik-g  thought of doing it, but should we tag this image which we are building manually as the latest?